### ISSUE TITLE
A huge number of data events

### ISSUE THREAD

**Issue by rndD** [kibae/pg-logical-replication#21]
Hello. Thanks for the package.
I use `pgoutput` to get the WAL log. 
Sometimes spike with huge amounts of updates happens and my pipeline process gets OOM because of the number of messages it received by the event emitter.

I thought I can `stop()` subscription after a specific limit of currently processing data events is reached. But I haven't found the `resume()` method.

What is the right way to limit the number of received WAL logs with the `pg-logical-replication`?

**Comment by kibae** [comment]
Hello, @rndD 👋
In nodejs' IO strategy model, an OOM error can occur if the delivery of changed data events is too fast than processing the delivered data.
I had the same issue a few years ago and since I had no control over the low level, I appended the changed data event information to a log file and then processed it by rereading the file line by line.
I hope this works for you.

https://github.com/kibae/pg-logical-replication/blob/9b191c5c4e0e9e953537c933fc331ae711e77587/src/logical-replication-service.ts#L111-L132
No matter what workaround we try to do in this area of code, the copyData events from the node-pg library may pile up and cause an OOM.

**Comment by rndD** [comment]
@kibae thanks for the response! I got it, it is impossible to implement on the `pg-logical-replication` level.

I don't think writing to FS will help me (don't have fast IO with the current setup). But thanks for the idea.


I tried to implement `resume()` method by myself. I simply run `service.subscribe(plugin, slotName)` again after the last batch of events is processed. And it works... But sometimes I got duplicate events after `resume()`. Events with the exact same `lsn` were already processed in the previous batch (but not all of them are duplicates).

Can it be due to events not getting acknowledged https://github.com/kibae/pg-logical-replication/blob/main/src/logical-replication-service.ts#L172 while service is stopped?

Probably I do not understand how pg works with lsn. 
If I start the service like that:
```
 await service.subscribe(new PgoutputPlugin({
        protoVersion: 1,
        publicationNames: publicationNames,
      }), slotName, lastLsn);
```

Does it guarantee that I will receive events only after  `lastLsn`? Apparently not.
